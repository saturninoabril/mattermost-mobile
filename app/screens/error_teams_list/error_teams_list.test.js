// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import FailedNetworkAction from 'app/components/failed_network_action';
import ErrorTeamsList from './error_teams_list';

describe('ErrorTeamsList', () => {
    const loadMe = async () => {
        return {
            data: {},
        };
    };

    const baseProps = {
        actions: {
            loadMe: jest.fn(),
            connection: jest.fn(),
            logout: jest.fn(),
            selectDefaultTeam: jest.fn(),
            resetToChannel: jest.fn(),
        },
        componentId: 'component-id',
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ErrorTeamsList {...baseProps}/>
        );

        expect(wrapper.find(FailedNetworkAction).exists()).toEqual(true);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should load user and select dafult team on getUserInfo', async () => {
        const wrapper = shallow(
            <ErrorTeamsList {...baseProps}/>
        );

        const instance = wrapper.instance();
        await instance.getUserInfo();

        expect(baseProps.actions.connection).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.connection).toBeCalledWith(true);

        expect(baseProps.actions.loadMe).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.loadMe).toBeCalledWith();

        expect(baseProps.actions.resetToChannel).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.resetToChannel).toBeCalledWith({disableTermsModal: true});

        expect(baseProps.actions.selectDefaultTeam).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.selectDefaultTeam).toBeCalledWith();

        expect(wrapper.state('loading')).toEqual(false);
    });
});
