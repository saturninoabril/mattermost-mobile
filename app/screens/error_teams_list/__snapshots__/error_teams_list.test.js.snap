// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTeamsList should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <FailedNetworkAction
    errorDescription={
      Object {
        "defaultMessage": "{type} will load when you have an internet connection or {refresh}.",
        "id": "mobile.failed_network_action.shortDescription",
        "values": Object {
          "refresh": <FormattedText
            defaultMessage="try again"
            id="mobile.failed_network_action.retry"
            onPress={[Function]}
            style={
              Object {
                "color": "#2389d7",
              }
            }
          />,
          "type": <FormattedText
            defaultMessage="Teams"
            id="mobile.failed_network_action.description.teams"
          />,
        },
      }
    }
    errorTitle={
      Object {
        "defaultMessage": "No internet connection",
        "id": "mobile.failed_network_action.title",
      }
    }
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;
